### PR TITLE
update Invoice entity

### PR DESCRIPTION
**Pull request by ankitdas13**
Links for api reference you can also check this [doc](https://docs.google.com/spreadsheets/d/1vF57ZRCCjkNdtyuGSbqwOWFaHPWHHgjN-faLzlrUS14/edit?usp=sharing) for more details

Invoice : https://razorpay.com/docs/api/payments/invoices

### CODE REVIEW

File: src/Invoice.cs [hunk 1]
```diff
@@ -11,9 +11,67 @@ public class Invoice : Entity
...
+            string relativeUrl = string.Format("{0}/{1}/{2}", GetUrlVersion(), GetEntityUrl(), this["id"]);
+            List<Entity> entities = Request(relativeUrl, HttpMethod.PATCH, data);
+            return (Invoice)entities[0];
+        }
+
+        public Invoice Cancel()
+        {
+            string relativeUrl = string.Format("{0}/invoices/{1}/cancel", GetUrlVersion(), this["id"]);
+            List<Entity> entities = Request(relativeUrl, HttpMethod.POST, null);
+            return (Invoice)entities[0];
```
Comment (ra0van) [inline src/Invoice.cs:41]: Check if array element with index 0 exists before directly fetching the value.

File: src/Invoice.cs [hunk 2]
```diff
@@ -11,9 +11,67 @@ public class Invoice : Entity
...
+        }
+
+        public Invoice NotifyBy(string medium)
+        {
+            string relativeUrl = string.Format("{0}/invoices/{1}/notify_by/{2}", GetUrlVersion(), this["id"], medium);
+            List<Entity> entities = Request(relativeUrl, HttpMethod.POST, null);
```
Comment (ra0van) [inline src/Invoice.cs:47]: Same here. Check if (Invoice)entities[0] exists first.

File: src/Invoice.cs (hunk 1 above)
Comment (ankitdas13) [reply src/Invoice.cs]: Add null check

File: src/Invoice.cs (hunk 2 above)
Comment (ankitdas13) [reply src/Invoice.cs:61]: Add null check

**Review by ra0van — APPROVED** [review]

**Pull request merged**